osquery/config/packs.cpp

```diff
@@ -216,7 +216,8 @@ const std::string& Pack::getVersion() const {
 }
 
 bool Pack::shouldPackExecute() {
-  return (valid_ && checkDiscovery());
+  executing_ = (valid_ && checkDiscovery());
+  return executing_.load();
```

Comment:
@theopolis, correct. We need to separate these things because we want the query scheduler to actually "do" the work, while the table should only access the cached result that the scheduler generated.